RevenueCatUI/Templates/Template7View.swift

```diff
@@ -0,0 +1,468 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Template7View.swift
+//
+//  Created by Nacho Soto.
+
+// swiftlint:disable type_body_length file_length
+
+import RevenueCat
+import SwiftUI
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct Template7View: TemplateViewType {
+
+    let configuration: TemplateViewConfiguration
+
+    private let tiers: [PaywallData.Tier: TemplateViewConfiguration.PackageConfiguration.MultiPackage]
+    private let tierNames: [PaywallData.Tier: String]
+
+    @State
+    private var selectedTier: PaywallData.Tier
+
+    @State
+    private var selectedPackage: TemplateViewConfiguration.Package
+
+    @State
+    private var displayingAllPlans: Bool
+
+    @Environment(\.userInterfaceIdiom)
+    var userInterfaceIdiom
+    @Environment(\.locale)
+    var locale
+
+    #if swift(>=5.9) || (!os(macOS) && !os(watchOS) && !os(tvOS))
+    @Environment(\.verticalSizeClass)
+    var verticalSizeClass
+    #endif
+
+    @Namespace
+    private var namespace
+
+    @EnvironmentObject
+    private var introEligibilityViewModel: IntroEligibilityViewModel
+    @EnvironmentObject
+    private var purchaseHandler: PurchaseHandler
+
+    init(_ configuration: TemplateViewConfiguration) {
+        guard let (firstTier, allTiers, tierNames) = configuration.packages.multiTier else {
+            fatalError("Attempted to display a multi-tier template with invalid data: \(configuration.packages)")
+        }
+
+        self.tiers = allTiers
+        self.tierNames = tierNames
+        self.configuration = configuration
+
+        self._selectedTier = .init(initialValue: firstTier)
+        self._selectedPackage = .init(initialValue: allTiers[firstTier]!.default)
```

Comment:
Can this be `firstTier.default`?

Reply:
`firtTier` is `PaywallData.Tier`. It's `default` is just a package identifier. This way we get the actual `TemplateViewConfiguration.Package`.